File: android/src/main/java/com/dlocal/plugins/datacollector/DataCollectorSDKPlugin.java
package com.dlocal.plugins.datacollector;

import com.dlocal.datacollector.api.DLEnvironment;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.dlocal.datacollector.DLCollector;
import com.dlocal.datacollector.api.DLSettings;
import com.dlocal.datacollector.api.DLEnvironment;

import org.json.JSONException;
import org.json.JSONObject;

@CapacitorPlugin(name = "DataCollectorSDK")
public class DataCollectorSDKPlugin extends Plugin {

    public DLEnvironment getEnv(Integer env) {
        switch (env) {
            case 0: return DLEnvironment.PRODUCTION;
            case 1: return DLEnvironment.SANDBOX;
        }
        return DLEnvironment.PRODUCTION;
    }

    @PluginMethod
    public void setUp(PluginCall call) {
        String apiKey = call.getString("apiKey");
        Integer env = call.getInt("env");
        DLEnvironment environment = env != null ? getEnv(env) : DLEnvironment.PRODUCTION;

        if (apiKey == null || apiKey.isEmpty()) {
            call.reject("Missing api key");
            return;
        }

        DLCollector.setUp(this.getContext(), new DLSettings(apiKey, environment));

        call.resolve();
    }

    @PluginMethod
    public void startSession(PluginCall call) { 
        DLCollector.getInstance().startSession();
        call.resolve();
    }

    @PluginMethod
    public void getSession(PluginCall call) {
        JSObject response = new JSObject();
        try {
            String id = DLCollector.getInstance().getSessionId();
            response.put("sessionId", id);
        } catch (Exception e) {
            response.put("sessionId", null);
        }
        call.resolve(response);
    }
}

File: ios/Plugin/DataCollectorSDKPlugin.swift
import Foundation
import Capacitor
import DLDataCollectorSDK

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(DataCollectorSDKPlugin)
public class DataCollectorSDKPlugin: CAPPlugin {
    private let implementation = DataCollectorSDK()
    
    @objc func setUp(_ call: CAPPluginCall) {
        let env = call.getInt("env")
        var environment: DLEnvironment = .production
        
        if let env = env, let dlEnv = DLEnvironment(rawValue: env) {
            environment = dlEnv
        }
        
        guard let apiKey = call.getString("apiKey")else {
                  call.reject("Missing api key")
                  return
              }

        DLCollector.shared.setUp(DLSettings(apiKey: apiKey,
                                            env: environment,
                                            logLevel: .silent))
    }
    
    @objc func startSession(_ call: CAPPluginCall) {
        do {
            try DLCollector.shared.startSession(additionalData: nil)
            call.resolve()
        } catch {
            call.reject("Error starting session \(error.localizedDescription)")
        }
    }
    
    @objc func getSession(_ call: CAPPluginCall) {
        let id = DLCollector.shared.getSessionId()
        call.resolve(["sessionId": id as Any])
    }
}
